client/src/App.js

```diff
@@ -1,44 +1,73 @@
-import Header from "./Header";
-import {useState} from 'react'
-import LoginForm from "./LoginForm";
-import RegisterForm from "./RegisterForm";
+import { useState, useEffect } from 'react'
+import Header from "./components/header/Header";
+import axios from 'axios';
+import LoginForm from "./components/LoginForm/LoginForm";
+import Dashboard from './components/Dashboard/Dashboard';
+import RegisterForm from "./components/RegisterForm/RegisterForm";
 
 function App() {
+
+  const [token, setToken] = useState();
   const [visibleLogin, setVisibleLogin] = useState(true)
   const [visibleRegister, setVisibleRegister] = useState(true)
-    const VisibleLogin = () => {
-      if (visibleLogin) {
-        return (
-          <div>
-          </div>
-        )
-      } else {
-          return (
-            <LoginForm/>
-          )
-      }   
+
+  const fetch = async () => {
+    await axios.post('/api/auth/authed', {
+      body: "hello"
+    })
+      .then(response => {
+        const token = response.data.success
+        setToken(token);
+      })
+  }
+
+  useEffect(() => {
+    fetch()
+  })
+
+  const VisibleLogin = () => {
+    if (visibleLogin) {
+      return (
+        <div>
+        </div>
+      )
+    } else {
+      return (
+        <LoginForm />
+      )
     }
+  }
 
-    const VisibleRegister = () => {
-      if (visibleRegister) {
-        return (
-          <div>
-          </div>
-        )
-      } else {
-          return (
-            <RegisterForm/>
-          )
-      }   
+  const VisibleRegister = () => {
+    if (visibleRegister) {
+      return (
+        <div>
+        </div>
+      )
+    } else {
+      return (
+        <RegisterForm />
+      )
     }
+  }
+
+  if (!token) {
+    return (
+      <div>
+        <Header setVisibleLogin={setVisibleLogin} visibleLogin={visibleLogin} setVisibleRegister={setVisibleRegister} visibleRegister={visibleRegister} />
+        <VisibleLogin />
```

Comment:
Just like I said above, this logic can be simplified to this:

```js
{visibleLogin && <LoginForm />}
```
And you can delete the VisibleLogin function